h2/src/main/org/h2/index/ViewIndex.java

```diff
@@ -285,6 +285,7 @@ private Cursor find(Session session, SearchRow first, SearchRow last,
             return findRecursive(first, last);
         }
         setupQueryParameters(session, first, last, intersection);
+        query.setNeverLazy(true);
```

Comment:
All the subqueries will become never-lazy now. Does not look like a correct fix.

Reply:
What do you think which subqueries should remain lazy and which shouldn't?

Reply:
Thanks. Fixed.

Reply:
There is a problem with CTEs. Take a look on Travis logs.

Reply:
@katzyn , thanks I see.
Fixed.

Reply:
Just curious: nested joins are intentionally not considered here?

Reply:
@katzyn, thanks. Looks like a bug. I'll investigate it.